bin/varnishtest/tests/bad_call.vtc

```diff
@@ -0,0 +1,34 @@
+varnishtest "VCL_SUB wrong vmod behavior"
+
+varnish v1 -vcl {
+	import debug;
+
+	backend dummy None;
+
+	sub foo {
+		set resp.http.it = "works";
+	}
+
+	sub vcl_init {
+		debug.bad_memory(foo);
```

Comment:
Shouldn't this be a compile error? You are invoking foo which touches resp in vcl_init?

Reply:
no, because it is not a `call`.
If that was an error, sub references were impossible.